mimic/model/nova_image_collection.py

```diff
@@ -6,32 +6,23 @@
 
 from characteristic import attributes, Attribute
 from json import dumps
-from mimic.model.rackspace_images import (ImageStore, OnMetalImage)
+from mimic.model.rackspace_images import OnMetalImage
 
 from mimic.model.nova_objects import not_found
 from mimic.canned_responses.mimic_presets import get_presets
 
 
 @attributes(
-    ["tenant_id", "region_name", "clock"])
-class RegionalImageCollection(object):
+    ["tenant_id", "region_name", "clock", "image_store"])
+class RegionalNovaImageCollection(object):
     """
-    A collection of images, in a given region, for a given tenant.
+    A collection of nova images, in a given region, for a given tenant.
     """
-    def image_by_id(self, image_id):
-        """
-        Retrieve a :obj:`Image` object by its ID.
-        """
-        images_store = ImageStore.create_image_store(self.tenant_id)
-        for image in images_store:
-            if image.image_id == image_id:
-                return image
-
     def list_images(self, include_details, absolutize_url):
         """
         Return a list of images.
         """
-        images_store = ImageStore.create_image_store(self.tenant_id)
+        images_store = self.image_store.create_image_store(self.tenant_id)
```

Comment:
Again, within `nova_image_collections`, a module named in a way that suggests it's a general-purpose thing for all Nova clouds, we are calling a method - `create_image_store` - that is only defined on a class within our rackspace-specific implementation.